codejam-canvas/src/js/index.js

```
@@ -0,0 +1,81 @@
+(function canvasModule() {
+  const dataSrc = {
+    "4": "4x4.json",
+    "32": "32x32.json"
+  };
+  const baseSize = 512;
+  const defaultSize = 4;
+
+  const errorMsg = document.querySelector(".error-msg");
+  let errorTimer;
+
+  const buttons = document.querySelectorAll(".aside-right__fsize");
+  const canvas = document.getElementById("canvas");
+  canvas.width = baseSize;
+  canvas.height = baseSize;
+  const ctx = canvas.getContext("2d");
+
+  function handleErrorMsg(action = "hide", time = 3000) {
+    if (action === "show") {
+      errorMsg.classList.add("shown");
+      errorTimer = setTimeout(() => {
+        errorMsg.classList.remove("shown");
+      }, time);
+    } else if (action === "hide") {
+      errorMsg.classList.remove("shown");
+      clearTimeout(errorTimer);
+    }
+  }
+
+  async function drawCanvas(size) {
+    const spriteSize = baseSize / size;
+    try {
+      const response = await fetch(`./assets/data/${dataSrc[size]}`);
+      if (!response.ok) {
+        throw new Error("Server response is not OK");
+      }
+      const data = await response.json();
+      data.forEach((row, i) => {
+        row.forEach((column, j) => {
+          ctx.fillStyle = Array.isArray(column)
+            ? `rgba(${column[0]},${column[1]},${column[2]},${column[3]})`
+            : `#${column}`;
+          ctx.fillRect(
+            spriteSize * i,
+            spriteSize * j,
+            spriteSize * (i + 1),
+            spriteSize * (j + 1)
+          );
+        });
+      });
+      handleErrorMsg("hide");
```

won't be error hidden automatically?

---

Error will be hidden automatically in 3 (by default) second if in this 3 sec other error won't be thrown. In other case, if error occurs again, message will may be hidden by previous setTimeout early. So we fired this function to stop setTimeout manualy.